### PR TITLE
Enhance Deposit Sidebar Views with New Components and Refactoring in Earn Protocol

### PR DESCRIPTION
**Pull request by marcinciarka**
This pull request includes various updates and refactorings to the `apps/earn-protocol` components to improve functionality and maintainability. The most important changes include the addition of new sidebar elements, removal of unused imports, and updates to component props.

### Component Enhancements:
* [`apps/earn-protocol/components/layout/VaultManageView/VaultManageViewComponent.tsx`](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL58-R56): Added `publicClient` to `useClient` and `useTransaction` hooks, and removed the `capitalize` function for the `title` prop in `sidebarProps`. [[1]](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL58-R56) [[2]](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL73-R73) [[3]](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL130-R129)
* [`apps/earn-protocol/components/layout/VaultOpenView/VaultOpenViewComponent.tsx`](diffhunk://#diff-af80e33b59ddc7605e3d3e1491d67643a7110bc68039c4d1cf0bac4460249f8fL63-R86): Added `publicClient` to `useClient` and `useTransaction` hooks, and refactored the sidebar content to include new `Approval`, `Deposit`, and `InitialDeposit` components. [[1]](diffhunk://#diff-af80e33b59ddc7605e3d3e1491d67643a7110bc68039c4d1cf0bac4460249f8fL63-R86) [[2]](diffhunk://#diff-af80e33b59ddc7605e3d3e1491d67643a7110bc68039c4d1cf0bac4460249f8fL108-L150)

### New Sidebar Elements:
* [`apps/earn-protocol/components/molecules/SidebarElements/Approval.tsx`](diffhunk://#diff-af3be6ef1c86e9eaafb815f54ce7754664cc9ebae1b6195ee1a9df5897b335d5R1-R81): Introduced a new `Approval` component to handle token approval interactions.
* [`apps/earn-protocol/components/molecules/SidebarElements/Deposit.tsx`](diffhunk://#diff-0e5512869e39797ee72b33de1c7793ef3531f48286d68d29918ea394181b6419R1-R37): Introduced a new `Deposit` component to display deposit details.
* [`apps/earn-protocol/components/molecules/SidebarElements/InitialDeposit.tsx`](diffhunk://#diff-108fe0034fae00e6ba3924143702cbc07930468b4dc01e4e80e87ceb7236584aR1-R61): Introduced a new `InitialDeposit` component to handle initial deposit interactions.

### Style Updates:
* [`apps/earn-protocol/components/molecules/SidebarElements/Approval.module.scss`](diffhunk://#diff-8939f0f734b35ca6922250d640aae5afa4b121a6a5adcdb51c3ffbd2266c4261R1-R48): Added new styles for the `Approval` component.
* [`apps/earn-protocol/components/molecules/SidebarElements/Deposit.module.scss`](diffhunk://#diff-f62d096e4aecf0884202bf8e7cd8aa0f71ed8a2f81c64334faa713081b3a0dbdR1-R31): Added new styles for the `Deposit` component.

### Code Cleanup:
* Removed unused imports from `VaultManageViewComponent.tsx` and `VaultOpenViewComponent.tsx` files to clean up the codebase. [[1]](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL4) [[2]](diffhunk://#diff-093401e3ee1390c1d36183263365baea6f0df465c9bc2cdafe8c6205c965caccL27) [[3]](diffhunk://#diff-af80e33b59ddc7605e3d3e1491d67643a7110bc68039c4d1cf0bac4460249f8fL3-L10) [[4]](diffhunk://#diff-af80e33b59ddc7605e3d3e1491d67643a7110bc68039c4d1cf0bac4460249f8fL24-L32)

## Release Notes

- **New Features**
  - Introduced `ControlsApproval`, `ControlsDepositWithdraw`, `OrderInfoDeposit`, and `OrderInfoWithdraw` components for enhanced transaction management.
  - Added `amountDisplayUSD` to display USD equivalent in the `useAmount` hook.
  - Updated sidebar functionality with a new "go back" action and improved layout.

- **UI Enhancements**
  - Updated icon names across various components to follow a new naming convention.
  - Enhanced styling for approval controls and deposit/withdraw information displays.
  - Improved layout and responsiveness of the sidebar component.

- **Bug Fixes**
  - Improved error handling and user interaction flows in transaction components.
  - Added suppression of hydration warnings in the `TableCellText` component.

- **Documentation**
  - Updated type definitions for new and modified components.

### CODE REVIEW

**Comment by marcinciarka** [comment]
@coderabbitai create a title for this PR

**Comment by marcinciarka** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> @marcinciarka: Your Majesty, may I humbly suggest the title **"Enhance Deposit Sidebar Views with New Components and Refactoring in Earn Protocol"** for this pull request?

Granted, now go back to your hole.

File: apps/earn-protocol/components/molecules/SidebarElements/Deposit.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Box, Icon, SimpleGrid, Text } from '@summerfi/app-earn-ui'
```
Comment (piekczyk) [inline apps/earn-protocol/components/molecules/SidebarElements/Deposit.tsx:1]: maybe it's worth to rename it to something like `DepositOrderInfo`.

File: apps/earn-protocol/features/portfolio/components/PortfolioAssets/PortfolioAssets.tsx [hunk 2]
```diff
@@ -19,19 +19,19 @@ const networks: { value: PortfolioAssetNetworkOption; icon?: ReactNode; label?:
...
   },
   {
     value: NetworkNames.optimismMainnet,
-    icon: <Icon iconName="network_optimism" variant="m" />,
+    icon: <Icon iconName="earn_network_optimism" variant="m" />,
   },
   {
     value: NetworkNames.arbitrumMainnet,
-    icon: <Icon iconName="network_arbitrum" variant="m" />,
+    icon: <Icon iconName="earn_network_arbitrum" variant="m" />,
```
Comment (piekczyk) [inline apps/earn-protocol/features/portfolio/components/PortfolioAssets/PortfolioAssets.tsx:34]: I think all other network icons beside `earn_network_arbitrum` looks quite the same. Check icons `network_base`, `network_ethereum` etc.

**Review by piekczyk — CHANGES_REQUESTED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**